src/components/dev-hub/tabs.js

```diff
@@ -0,0 +1,146 @@
+        :after {
+            background-color: ${({ theme }) => theme.colorMap.darkGreen};
+        }
+    }
+`;
+
+const SLUG_TO_STRING = {
+    shell: 'Mongo Shell',
+    compass: 'Compass',
+    python: 'Python',
+    javasync: 'Java (Sync)',
+    'java-sync': 'Java (Sync)',
+    nodejs: 'Node.js',
+    php: 'PHP',
+    motor: 'Motor',
+    'java-async': 'Java (Async)',
+    c: 'C',
+    cpp: 'C++11',
+    csharp: 'C#',
+    perl: 'Perl',
+    ruby: 'Ruby',
+    scala: 'Scala',
+    go: 'Go',
+    cloud: 'Cloud',
+    local: 'Local',
+    macos: 'macOS',
+    linux: 'Linux',
+    windows: 'Windows',
+    debian: 'Debian',
+    rhel: 'RHEL',
+};
+
+const hiddenTabStyling = css`
+    & > div:first-child {
+        display: none;
+    }
+`;
+
+const StyledTabs = styled(LeafyTabs)`
+    ${({ ishidden }) => ishidden && hiddenTabStyling};
+    button[aria-selected='true'] {
+        color: ${({ theme }) => theme.colorMap.darkGreen};
+        :after {
+            background-color: ${({ theme }) => theme.colorMap.darkGreen};
+        }
+    }
+`;
+
+const stringifyTab = tabName => SLUG_TO_STRING[tabName] || tabName;
+
+const getTabId = node => getNestedValue(['options', 'tabid'], node);
+
+// Name anonymous tabsets by alphabetizing their tabids and concatenating with a forward slash
+const generateAnonymousTabsetName = tabIds => [...tabIds].sort().join('/');
+
+const TabButton = ({ ...props }) => (
+    <StyledTabButton renderAsButton {...props} />
+);
+
+const Tabs = ({ nodeData: { children, options = {} } }) => {
+    const hidden = options.hidden;
+    const { activeTabs, setActiveTab } = useContext(TabContext);
+    const tabIds = children.map(child => getTabId(child));
+    const tabsetName = options.tabset || generateAnonymousTabsetName(tabIds);
+    const [activeTab, setActiveTabIndex] = useState(0);
+    const previousTabsetChoice = activeTabs[tabsetName];
+    const tabs = children;
+
+    useEffect(() => {
+        if (!previousTabsetChoice || !tabIds.includes(previousTabsetChoice)) {
+            // Set first tab as active if no tab was previously selected
+            setActiveTab({ tabset: tabsetName, value: getTabId(children[0]) });
+        }
+    }, []); // eslint-disable-line react-hooks/exhaustive-deps
+
+    useEffect(() => {
+        const index = tabIds.indexOf(activeTabs[tabsetName]);
+        if (index !== -1) {
+            setActiveTabIndex(index);
+        }
+    }, [activeTabs, tabIds, tabsetName]);
+
+    const onClick = useCallback(
+        index => {
+            setActiveTab({
+                tabset: tabsetName,
+                value: getTabId(children[index]),
+            });
+        },
+        [children, setActiveTab, tabsetName]
+    );
+
+    return (
+        <>
+            <StyledTabs
+                selected={activeTab}
+                setSelected={onClick}
+                darkMode
+                ishidden={hidden}
+                as={TabButton}
```

Comment:
Another issue was with this `as` prop. When using the default `button` value, the page would jump on a re-render to the last set of tabs. Sophie and I have no idea what is the root cause, but using anything else here, such as our `Button` does not give this issue so I went ahead and swapped our button in here.